### PR TITLE
[refactor][timer] - timer-context for POMODORO

### PR DESCRIPTION
**Pull request by abhij1607**
- Added timer-context to allow manipulate time for Pomodoro and Break by user in upcoming feature
- refactored code to use timer-context in pomodoro-pane

### CODE REVIEW

File: src/components/pomodoro-pane.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+    pause,
+    pomodoroTimer,
+    shortBreakTimer,
+    longBreakTimer,
+    menuOptionActive,
+  } = timerState;
+  const [timer, setTimer] = useState(pomodoroTimer);
 
   const tick = () => {
     setTimer(timer - 1);
```
Comment (shivampandey0) [inline src/components/pomodoro-pane.js]: Do not mutate state.
```suggestion
      setTimer(prev=> prev-1);
```

**Review by junaidshaikh-js — APPROVED** [review]
Good use of context for the Pomodoro clock

Can be merged.

File: src/components/pomodoro-pane.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
 import { useEffect, useState } from "react";
+import { useTimer } from "../context/timer-context";
 
 const PomodoroPane = () => {
-  const [timer, setTimer] = useState(1500);
-  const [pause, setPause] = useState(true);
-  const [menuOptionActive, setMenuOptionActive] = useState("Pomodoro");
+  const { timerState, timerDispatch } = useTimer();
+  const {
+    pause,
```
Comment (shivampandey0) [inline src/components/pomodoro-pane.js]: You can make this state locally where you are using this.

**Review by shivampandey0 — COMMENTED** [review]
Easy to understand code with semantic HTML tags.
- Added some suggestions

File: src/components/pomodoro-pane.js (hunk 2 above)
Comment (abhij1607) [reply src/components/pomodoro-pane.js]: done, thanks for the suggestion

File: src/components/pomodoro-pane.js (hunk 1 above)
Comment (abhij1607) [reply src/components/pomodoro-pane.js]: done, thanks for the suggestion

**Pull request merged**